### PR TITLE
Include lessons from other projects.

### PR DESCRIPTION
**Pull request by izaakschroeder**
- Include `-loader` in the request to avoid loading wrong modules,
- Use CSS modules for `postcss`,
- Support for `SOURCE_VERSION` instead of just using `git`,
- Remove `dotenv`; let the user handle their environment,
- Improved HMR support.

/cc @jjt

### CODE REVIEW

File: generators/postcss/templates/postcss.webpack.config.js [hunk 1]
```diff
@@ -1,58 +1,77 @@
...
 import autoprefixer from 'autoprefixer-core';
 import precss from 'precss';
 import lost from 'lost';
 import cssimport from 'postcss-import';
-import csswring from 'csswring';
+import fonts from 'postcss-font-magician';
 
 const IS_STYLE = /\.(scss|sass|css)$/;
 
+function q(loader, query) {
```
Comment (jjt) [inline generators/postcss/templates/postcss.webpack.config.js]: `query` too long?

File: generators/postcss/templates/postcss.webpack.config.js [hunk 2]
```diff
@@ -1,58 +1,77 @@
...
-      // Some crawlers or things with Javascript disabled prefer normal CSS
-      // instead of Javascript injected CSS, so this plugin allows for the
-      // collection of the generated CSS into its own file.
-      new ExtractTextPlugin('[name].[contenthash].css')
-    ] : [ ])
+    plugins: [
+      ...(target !== 'web' ? [
+        // Ignore CSS far as server-side goes
+
+      ] : [ ]),
```
Comment (jjt) [inline generators/postcss/templates/postcss.webpack.config.js]: I find the empty/null path up front is easier on the eyes:

``` js
...(target === 'web' ? [] :
  [
    // Ignore CSS as far as server-side goes
  ]),
```

**Comment by jjt** [comment]
Minor line notes, LGTM

File: generators/postcss/templates/postcss.webpack.config.js (hunk 1 above)
Comment (izaakschroeder) [inline generators/postcss/templates/postcss.webpack.config.js]: Good call lol.

Comment (jjt) [inline generators/postcss/templates/postcss.webpack.config.js]: ;)

**Pull request merged**